thoth-api/src/graphql/model.rs

```diff
@@ -1970,12 +1975,35 @@ impl MutationRoot {
         #[graphql(description = "Values to apply to existing location")] data: PatchLocation,
     ) -> FieldResult<Location> {
         context.token.jwt.as_ref().ok_or(ThothError::Unauthorised)?;
-        let location = Location::from_id(&context.db, &data.location_id).unwrap();
+        let current_location = Location::from_id(&context.db, &data.location_id).unwrap();
+        let publication = Publication::from_id(&context.db, &data.publication_id).unwrap();
```

Comment:
you can simplify all this by concatenating the queries:

```
let has_canonical_thoth_location = Publication::from_id(&context.db, &data.publication_id)?
            .locations(
                context,
                Some(1),
                None,
                None,
                Some(vec![LocationPlatform::Thoth]),
            )?
            .first()
            .map_or(false, |location| location.canonical);
```

Reply:
Thank you! I knew that this part could be simplified.